File: dev-docs/snippets/workflows/get-status.swift
import JsBaoClient

// Poll the status of a run.
func getStatus(client: JsBaoClient, workflowKey: String, runKey: String) async throws {
  // #region example
  let status = try await client.workflows.getStatus(
    workflowKey: workflowKey,
    runKey: runKey
  )
  // status.status / status.output / status.error / status.run
  // #endregion example
  _ = status
}

File: dev-docs/snippets/workflows/get-status.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Poll the status of a run. Omit contextDocId to use the user's root document.
export async function getStatus(
  client: JsBaoClient,
  workflowKey: string,
  runKey: string,
) {
  // #region example
  const status = await client.workflows.getStatus({ workflowKey, runKey });
  // status.status / status.output / status.error / status.run
  // #endregion example
  return status;
}
